apps/worker/src/app/workflow/usecases/run-job/run-job.usecase.ts

```diff
@@ -40,6 +43,14 @@ export class RunJob {
     let job = await this.jobRepository.findOne({ _id: command.jobId, _environmentId: command.environmentId });
     if (!job) throw new PlatformException(`Job with id ${command.jobId} not found`);
 
+    const template = await this.getNotificationTemplate({
+      _id: job._templateId,
+      environmentId: job._environmentId,
+    });
+    if (!template) {
+      throw new PlatformException(`Notification template ${job._templateId} is not found`);
+    }
```

Comment:
Find the `template` when the job is running and propagate it's `tags` to the place where the `message` document is created